frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-dropdown.tsx

```diff
@@ -0,0 +1,107 @@
+import { NooBaaNamespaceStoreModel } from '../../models';
+import { NamespaceStoreKind } from '../../types';
+
+export const NamespaceStoreDropdown: React.FC<NamespaceStoreDropdownProps> = ({
+  namespace,
+  onChange,
+  className,
+}) => {
+  const { t } = useTranslation();
+  const [isOpen, setOpen] = React.useState(false);
+  const [nnsObj, nnsLoaded, nnsErr] = useK8sGet<ListKind<NamespaceStoreKind>>(
+    NooBaaNamespaceStoreModel,
+    null,
+    namespace,
+  );
+  const [nsName, setNSName] = React.useState('');
+  const [dropdownItems, setDropdownItems] = React.useState([]);
+  const nnsList = nnsLoaded ? nnsObj.items : [];
+  const handleDropdownChange = React.useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      setNSName(e.currentTarget.id);
+      onChange(nnsList.find((nns) => nns?.metadata?.name === e.currentTarget.id));
+    },
+    [nnsList, onChange],
+  );
+  React.useEffect(() => {
+    const nnsDropdownItems = nnsList.reduce(
+      (res, nns) => {
+        res.push(
+          <DropdownItem
+            key={nns.metadata.uid}
+            component="button"
+            id={nns?.metadata?.name}
+            onClick={handleDropdownChange}
+            data-test={nns?.metadata?.name}
+            description={t('ceph-storage-plugin~Provider {{provider}} | Region: {{region}}', {
+              provider: nns?.spec?.type,
+              region: nns?.spec?.awsS3?.region,
+            })}
+          >
+            {nns?.metadata?.name}
+          </DropdownItem>,
+        );
+        return res;
+      },
+      [
+        <DropdownItem
+          data-test="create-new-namespacestore-button"
+          key="first-item"
+          component="button"
+          onClick={() => NamespaceStoreModal({ namespace })}
+        >
+          {t('ceph-storage-plugin~Create New Namespace Store')}
+        </DropdownItem>,
+        <DropdownSeparator key="separator" />,
+      ],
+    );
+    setDropdownItems(nnsDropdownItems);
+  }, [nnsObj, nnsLoaded, nnsErr, nnsList, t, handleDropdownChange, namespace]);
+
+  return (
+    <div className={className}>
+      {nnsErr && (
+        <Alert
+          className="nb-create-bc-step-page__danger"
+          variant="danger"
+          isInline
+          title="An error has occured while fetching namespace stores"
+        />
+      )}
+      <Dropdown
+        className="dropdown dropdown--full-width"
+        toggle={
+          <DropdownToggle
+            id="nns-dropdown-id"
+            data-test="nns-dropdown-toggle"
+            onToggle={() => setOpen(!isOpen)}
+          >
+            {nsName || t('ceph-storage-plugin~Select a Namespace Store')}
+          </DropdownToggle>
+        }
+        isOpen={isOpen}
+        dropdownItems={dropdownItems}
+        onSelect={() => setOpen(false)}
+        id="noobaa-namespace-store"
+      />
+    </div>
+  );
+};
+
+type NamespaceStoreDropdownProps = {
+  namespace?: string;
```

Comment:
I see you require namspace for fething some data above.
Why optional ?

Reply:
changed it.